agent/tool-scripts/base-tool

```diff
@@ -590,39 +591,33 @@ install)
 		fi
 		return ${rc}
 	}
-	case "${tool}" in
-	bpftrace|haproxy-ocp|iostat|kvmtrace|mpstat|oc|perf|pidstat|prometheus-metrics|sar|turbostat)
+	if [[ ! -z "${tool_package_name}" ]]; then
 		check_required_rpm ${tool_package_name} ${tool_package_ver}
 		rc=${?}
-		;;
-	pprof)
-		check_required_rpm ${tool_package_name} ${tool_package_ver}
-		rc=${?}
-		if [[ ${rc} == 0 ]]; then
-			# For pprof, we also need ansible installed.
-			check_required_rpm "ansible"
-			rc=${?}
-		fi
-		;;
-	systemtap)
-		# We need a number of systemtap and kernel packages for the
-		# systemtap tool to actually work.
-		check_required_rpm "systemtap-client"
-		rc=${?}
-		if [[ ${rc} == 0 ]]; then
-			check_required_rpm "systemtap-devel"
-			rc=${?}
-		fi
 		if [[ ${rc} == 0 ]]; then
-			check_required_rpm "kernel-devel" "$(uname -r)"
-			rc=${?}
-		fi
-		if [[ ${rc} == 0 ]]; then
-			check_required_rpm "kernel-debuginfo" "$(uname -r)"
-			rc=${?}
+			case "${tool}" in
+			pprof)
+				# We also need ansible installed.
+				check_required_rpm "ansible"
+				rc=${?}
+				;;
+			systemtap)
```

Comment:
Another solution might have been to let `tool_package_name` become a list, where you'd check each package, rather than special casing on the `tool` name.

It bothers me just a little that this special casing for the tool is inside the `! -z "${tool_package_name}"` lest anyone see the case and try to leverage it to special-case another tool without adding `tool_package_name` for that tool. It's a minor concern, and I wouldn't want it to stand in the way here ... but doing the list would be a more general solution.

(Well, OK, not quite trivial as you need to know where to use the `$(uname -r)` ... since `require-rpm` would do the wrong thing if that's supplied where not necessary. ☹️ OK, so, forget that; it's better than it was, we're good.)

Reply:
Yes, I debated this as well.

I felt that with `systemtap` being the only tool that has so many packages required, implementing a general mechanism would not solve much.  By adding `pprof` to now have two (`golang` and `ansible`), i considered that might tip it in favor of a general solution.  But then I asked myself, "do I really want to continue this solution in `bash`?"

So for now, I figured we just have 2 special cases (`pprof` and `systemtap`) and an ugly bit of indentation instead.